smartype-api/src/commonMain/kotlin/com/mparticle/smartype/api/SmartypeApi.kt

```diff
@@ -1,27 +1,23 @@
 package com.mparticle.smartype.api
 
+import kotlin.js.JsExport
+import kotlin.js.JsName
 
-import kotlinx.serialization.json.Json
-import kotlinx.serialization.json.JsonObject
-import kotlinx.serialization.modules.SerializersModule
-import kotlinx.serialization.modules.SerializersModuleCollector
-import kotlinx.serialization.modules.SerializersModuleBuilder
-
-public abstract class SmartypeApiBase(private val receivers: kotlin.collections.List<MessageReceiver>) {
-    private val mutableReceivers = receivers.toMutableList()
-
+@JsExport
+public abstract class SmartypeApiBase() {
+    private val mutableReceivers = kotlin.collections.mutableListOf<MessageReceiver>()
 
+    @JsName("send")
     public fun send(message: Message) {
-        val json = Json { isLenient = true }
-
-        mutableReceivers.forEach { it.receive(json.encodeToJsonElement(Message.serializer(), message) as JsonObject) }
+        mutableReceivers.forEach { it.receive(message.toJson()) }
```

Comment:
You should be able to call the toJson before the loop to save extra serialization

Reply:
Great suggestion! This is implemented in latest push (as well as your other feedback)